agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/DefaultClassNamesProvider.java

```diff
@@ -0,0 +1,215 @@
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.agent;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+
+import com.microsoft.applicationinsights.agent.internal.config.AgentConfiguration;
+import com.microsoft.applicationinsights.agent.internal.coresync.InstrumentedClassType;
+import com.microsoft.applicationinsights.agent.internal.logger.InternalAgentLogger;
+
+/**
+ * Created by gupele on 5/11/2015.
+ */
+class DefaultClassNamesProvider implements ClassNamesProvider {
+    private AgentConfiguration agentConfiguration;
+
+    private final HashSet<String> sqlClasses = new HashSet<String>();
+    private final HashSet<String> httpClasses = new HashSet<String>();
+    private final HashSet<String> forbiddenPaths = new HashSet<String>();
+
+    private final HashMap<String, ClassInstrumentationData> classesToInstrument = new HashMap<String, ClassInstrumentationData>();
+
+    private boolean builtInEnabled = true;
+
+    public DefaultClassNamesProvider() {
+        populateForbiddenClasses();
+    }
+
+    @Override
+    public void setConfiguration(AgentConfiguration agentConfiguration) {
+        this.agentConfiguration = agentConfiguration;
+        addConfigurationData();
+
+        if (builtInEnabled) {
+            InternalAgentLogger.INSTANCE.trace("Adding built-in instrumentation");
+
+            populateSqlClasses();
+            populateHttpClasses();
+            addHibernate();
+        }
+    }
+
+    @Override
+    public boolean isSqlClass(String className) {
+        return sqlClasses.contains(className);
+    }
+
+    @Override
+    public boolean isHttpClass(String className) {
+        return httpClasses.contains(className);
+    }
+
+    /**
+     * Gets the {@link ClassInstrumentationData} that is associated
+     * with the argument 'className', and removes that entry from the container once this is found
+     * @param className The class name to search for
+     * @return The {@link ClassInstrumentationData}
+     */
+    @Override
+    public ClassInstrumentationData getAndRemove(String className) {
+        return classesToInstrument.remove(className);
+    }
+
+    private void addHibernate() {
+        HashSet<String> methodNamesOnly = new HashSet<String>();
+        methodNamesOnly.add("delete");
```

Comment:
I think it would be more elegant here and below defining a string array with the method names, forbidden paths, etc. and then call the relevant methods on the relevant string arrays in a loop.